src/json.jl

```diff
@@ -1,17 +1,14 @@
 # -------------------------------- #
 # Custom JSON output for our types #
 # -------------------------------- #
-JSON._print(io::IO, state::JSON.State, a::HasFields) =
-    JSON._print(io, state, a.fields)
+JSON.lower(a::HasFields) = a.fields
 
-JSON._print(io::IO, state::JSON.State, d::Base.Dates.Date) =
-    JSON._print(io, state, "$(Base.Dates.year(d))-$(Base.Dates.month(d))-$(Base.Dates.day(d))")
+JSON.lower(d::Base.Dates.Date) =
```

Comment:
Ah, I see that this feature was introduced only starting v0.5.1. Also, this version will serialize dates as `2016-8-19`—the JSON version is `2016-08-19`. I think the latter would probably work in your case too, in which case this line can be safely deleted.